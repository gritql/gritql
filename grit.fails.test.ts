//@ts-ignore
const { gqlToDb, gqlBuild } = require('./gql-sql-slicer')
describe('SQL', () => {
  describe('failed with $#@# instead of .', () => {
    test('should correctly return names with dots inside like test.com.us', () => {
      const table = [
        [
          {
            domain: 'test.com',
            value: '101',
          },
          {
            domain: 'test2.com',
            value: '107',
          },
          {
            domain: 'test.com.us',
            value: '201',
          },
          {
            domain: 'test.com.ua',
            value: '207',
          },
        ],
      ]

      const querier = gqlToDb()
        .beforeDbFetch(({ sql }) => {
          expect(sql).toMatchSnapshot()
          return { sql }
        })
        .dbFetch(() => {
          return table
        })

      return Promise.resolve(
        querier(
          `input Date {
          gte: String!
          lte: String!
        }
        
        input Date {
          gte: String!
          lte: String!
        }
        
        input OrArray {
          in: [String]
        }
        
        input Filters {
          country: String
          category: OrArray
          domain: OrArray!
          date: Date!
        }
        
        query ii_transactional($filters: Filters) {
          matrix: fetch(filters: $filters) {
            domain(type:Array, sort_desc: revenue, limit: 10){
              value: sum(a:transactionrevenue)
            }
          }
        }`,
          {
            filters: {
              country: 'US',
              domain: {
                in: ['test.com', 'test2.com'],
              },
              date: {
                gte: '2022-01-05',
                lte: '2022-01-08',
              },
            },
          },
        ),
      ).then((result) => {
        expect(result).toMatchSnapshot()
      })
    })
  })
  describe('doesnt merge queries with same id from different queries', () => {
    test('should merge object correctly if id matches', () => {
      const table = [
        [
          {
            date: '1',
            value: '101',
          },
          {
            date: '2',
            value: '107',
          },
          {
            date: '3',
            value: '201',
          },
          {
            date: '4',
            value: '207',
          },
        ],
        [
          {
            date: '1',
            value1: '1011',
          },
          {
            date: '2',
            value1: '1071',
          },
          {
            date: '3',
            value1: '2011',
          },
          {
            date: '4',
            value1: '2071',
          },
        ],
      ]

      const querier = gqlToDb()
        .beforeDbFetch(({ sql }) => {
          expect(sql).toMatchSnapshot()
          return { sql }
        })
        .dbFetch(() => {
          return table
        })

      return Promise.resolve(
        querier(
          `input Date {
          gte: String!
          lte: String!
        }
        
        input Date {
          gte: String!
          lte: String!
        }
        
        input OrArray {
          in: [String]
        }
        
        input Filters {
          country: String
          category: OrArray
          domain: OrArray!
          date: Date!
        }
        
        query ii_transactional($filters: Filters) {
          matrix: fetch(filters: $filters) {
            date(type:Array, sort_desc: revenue, limit: 10){
              value: sum(a:transactionrevenue)
            }
          }

        }
        query ii_transactionalb($filters: Filters) {
          matrix: fetch(filters: $filters) {
            date(type:Array, sort_desc: revenue, limit: 10){
              value1: sum(a:transactionrevenue)
            }
          }
        }`,
          {
            filters: {
              country: 'US',
              domain: {
                in: ['test.com', 'test2.com'],
              },
              date: {
                gte: '2022-01-05',
                lte: '2022-01-08',
              },
            },
          },
        ),
      ).then((result) => {
        expect(result).toMatchSnapshot()
      })
    })
  })
})

xdescribe('directive', () => {
  describe('parse doesnt work for date field', () => {
    test('should correctly parse date', () => {
      const table = [
        [
          {
            date: 'Thu Jan 06 2022 01:00:00 GMT+0100 (Central European Standard Time)',
            value: '101',
          },
          {
            date: 'Thu Jan 07 2022 01:00:00 GMT+0100 (Central European Standard Time)',
            value: '102',
          },
          {
            date: 'Thu Jan 08 2022 01:00:00 GMT+0100 (Central European Standard Time)',
            value: '103',
          },
        ],
      ]

      const querier = gqlToDb()
        .beforeDbFetch(({ sql }) => {
          expect(sql).toMatchSnapshot()
          return { sql }
        })
        .dbFetch(() => {
          return table
        })

      return Promise.resolve(
        querier(
          `input Date {
          gte: String!
          lte: String!
        }
        
        input Date {
          gte: String!
          lte: String!
        }
        
        input OrArray {
          in: [String]
        }
        
        input Filters {
          country: String
          category: OrArray
          domain: OrArray!
          date: Date!
        }
        
        query ii_transactional($filters: Filters) {
          matrix: fetch(filters: $filters) {
            date(type:Array, sort_desc: date) @parse(as: "date:iso"){
              value: sum(a:transactionrevenue)
            }
          }
        }`,
          {
            filters: {
              country: 'US',
              domain: {
                in: ['test.com', 'test2.com'],
              },
              date: {
                gte: '2022-01-05',
                lte: '2022-01-08',
              },
            },
          },
        ),
      ).then((result) => {
        expect(result).toMatchSnapshot()
      })
    })
  })
})
