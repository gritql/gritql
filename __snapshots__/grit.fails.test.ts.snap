// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SQL doesnt merge queries with same id from different queries should merge object correctly if id matches 1`] = `
Array [
  "select \\"date\\", sum(\\"transactionrevenue\\") as \\"value\\" from \\"ii_transactional\\" where (\\"ii_transactional\\".\\"country\\" = 'US') AND ((\\"ii_transactional\\".\\"domain\\" IN ('test.com','test2.com'))) AND ((\\"ii_transactional\\".\\"date\\" >= '2022-01-05') AND (\\"ii_transactional\\".\\"date\\" <= '2022-01-08')) group by \\"date\\" order by \\"revenue\\" desc limit 10",
  "select \\"date\\", sum(\\"transactionrevenue\\") as \\"value1\\" from \\"ii_transactionalb\\" where (\\"ii_transactionalb\\".\\"country\\" = 'US') AND ((\\"ii_transactionalb\\".\\"domain\\" IN ('test.com','test2.com'))) AND ((\\"ii_transactionalb\\".\\"date\\" >= '2022-01-05') AND (\\"ii_transactionalb\\".\\"date\\" <= '2022-01-08')) group by \\"date\\" order by \\"revenue\\" desc limit 10",
]
`;

exports[`SQL doesnt merge queries with same id from different queries should merge object correctly if id matches 2`] = `
Object {
  "matrix": Array [
    Object {
      "date": "1",
      "value": "101",
      "value1": "1011",
    },
    Object {
      "date": "2",
      "value": "107",
      "value1": "1071",
    },
    Object {
      "date": "3",
      "value": "201",
      "value1": "2011",
    },
    Object {
      "date": "4",
      "value": "207",
      "value1": "2071",
    },
  ],
}
`;

exports[`SQL failed with $#@# instead of . should correctly return names with dots inside like test.com.us 1`] = `
Array [
  "select \\"domain\\", sum(\\"transactionrevenue\\") as \\"value\\" from \\"ii_transactional\\" where (\\"ii_transactional\\".\\"country\\" = 'US') AND ((\\"ii_transactional\\".\\"domain\\" IN ('test.com','test2.com'))) AND ((\\"ii_transactional\\".\\"date\\" >= '2022-01-05') AND (\\"ii_transactional\\".\\"date\\" <= '2022-01-08')) group by \\"domain\\" order by \\"revenue\\" desc limit 10",
]
`;

exports[`SQL failed with $#@# instead of . should correctly return names with dots inside like test.com.us 2`] = `
Object {
  "matrix": Array [
    Object {
      "domain": "test.com",
      "value": "101",
    },
    Object {
      "domain": "test2.com",
      "value": "107",
    },
    Object {
      "domain": "test.com.us",
      "value": "201",
    },
    Object {
      "domain": "test.com.ua",
      "value": "207",
    },
  ],
}
`;

exports[`directive parse doesnt work for date field should correctly parse date 1`] = `
Array [
  "select \\"date\\", sum(\\"transactionrevenue\\") as \\"value\\" from \\"ii_transactional\\" where (\\"ii_transactional\\".\\"country\\" = 'US') AND ((\\"ii_transactional\\".\\"domain\\" IN ('test.com','test2.com'))) AND ((\\"ii_transactional\\".\\"date\\" >= '2022-01-05') AND (\\"ii_transactional\\".\\"date\\" <= '2022-01-08')) group by \\"date\\" order by \\"date\\" desc",
]
`;

exports[`directive parse doesnt work for date field should correctly parse date 2`] = `
Object {
  "matrix": Array [
    Object {
      "date": "Thu Jan 06 2022 01:00:00 GMT+0100 (Central European Standard Time)",
      "value": 101,
    },
    Object {
      "date": "Thu Jan 07 2022 01:00:00 GMT+0100 (Central European Standard Time)",
      "value": 101,
    },
    Object {
      "date": "Thu Jan 08 2022 01:00:00 GMT+0100 (Central European Standard Time)",
      "value": 101,
    },
  ],
}
`;
